### DISCUSSION TITLE
Need help with your coding test? Start here.

### DISCUSSION THREAD

**Discussion by fenske** [DevSkillsHQ/discusssions#3] [Help]
Hi there! 👋  Guessing you're here because you've run into a problem with your coding test. Happy to assist!

Please check out the posts below as they tackle the most commonly asked questions:

- [I get a 404 when clicking on my repo link](https://www.notion.so/devskills/I-get-a-404-when-clicking-on-my-repo-link-091dcf187f7d48b8a69edfea4b539552)
- [I can't clone my assignment repository](https://www.notion.so/devskills/I-can-t-clone-my-assignment-repository-749b58bbd7d9449a91aa3a091bcf6590)
- [What tech stack am I allowed to use?](https://www.notion.so/devskills/What-tech-stack-am-I-allowed-to-use-4cfde6b2119a4546909db91f451e55e2)
- [I don't know how to make the tests run in the pipeline](https://www.notion.so/devskills/I-don-t-know-how-to-make-the-tests-run-in-the-pipeline-31cddf1afd7e463a899b97c2470c5e76)
- [Tests pass locally, but fail in the pipeline](https://www.notion.so/devskills/Tests-pass-locally-but-fail-in-the-pipeline-95aab1cbf6c44b56aed5b07d1df5360f)
- [Can't run the tests locally](https://www.notion.so/devskills/Can-t-run-the-tests-locally-52ffa2aa03994dadbe1166e68e3a7b11)

But if that doesn't help, please create a new discussion with your question by clicking [**this link**](https://github.com/orgs/DevSkillsHQ/discussions/new?category=help) to get assistance.

**Comment by ghofranebenhmaid** [comment]


**Reply by fenske** [reply]
Hi @ghofranebenhmaid .  As far as I know, we don't have a vite+react boilerplate on the list (only vue+vite & react+nextjs). 🤔 

Anyway, it doesn't seem like it's configured according to the instructions, as the build & start commands in package.json are still not set up. 

Please double-check the content of the "Import boilerplate" section.

**Reply by fenske** [reply]
@ghofranebenhmaid I imported the react boilerplate and pushed it to the [implementation](https://github.com/93c6399/transaction-management-frontend-level-1_c9b32af-4543hb/tree/implementation) branch of your repo.

Now it's just about adding the implementation. 

Hope this helps.

Best,
Anton.

**Comment by jclopes** [comment]
I had to explicitly install 'npm' and 'cypress'. There where no instructions to do this on the setup instructions.
I'm using arch linux.

**Reply by fenske** [reply]
Hi @jclopes . Anton from DevSkills is here.

You didn't have to. The README lists 2 other options. The one that requires NPM mentions the prereqs (NPM is part of the node package and Cypress is installed automatically once you run `npm install`).

Copy-pasting a screenshot from your README for clarity:

<img width="909" alt="CleanShot 2022-11-12 at 13 15 44@2x" src="https://user-images.githubusercontent.com/1162212/201473516-9267c7a2-1866-48fe-b931-290f16ef608e.png">

Let me know if you have any other questions.

Best,
Anton.

**Comment by alitanharo** [comment]
Hi, 
I am using fullstack-boilerplate-express-react  for this test and my app works 100% perfectly.  I have a problem running tests, I received this error
: The process '/usr/local/bin/npm' failed with exit code 1
I can't understand what the problem is, already tried all the previous solutions that exist in your community but they still don't work.
Can you Guide me, please?

**Reply by fenske** [reply]
Hi @alitanharo. I'm Anton from Alva. Happy to help. This is the root cause. 👇  ![CleanShot 2023-05-13 at 22 33 55@2x](https://github.com/DevSkillsHQ/discusssions/assets/71498485/b7df7b7e-144c-4d2f-8cd3-249c0184fa72)

I.e. the build command wouldn't run successfully due to a compiling issue.

Hope this helps.

**Reply by alitanharo** [reply]
Hi Anton,
Thank you for your response,
It was helpful!

**Comment by simepavlic** [comment]
Hello,
It is not clear to me how current poll and next polls are chosen. There is a presentation property "current_poll_index", but I have found no explanation how it is indexed(from 0, or from 1). When I give my polls indices in the way they are ordered in the initial POST /presentations request, I don't get the expected results when I fetch these polls based on "current_poll_index" property.
I'm not sure if I'm speaking clearly, but would love some help on this issue

**Reply by fenske** [reply]
Hi there. If I understood your question correctly, you pretty much want to know what GET /polls/current would return right after a presentation is created. To answer that, you could check out the "creating a presentation, showing a poll, voting and reading votes" test case which calls PUT /polls/current and subsequently validates the response content.

**Comment by EmranAhmed00** [comment]
Hi  ,

 I would to ask how can I submit my solutions ! . Can i share the Pull request links or can i add  any collaborators  if you provide any user name /team.

Many Thanks ,
Regards,

**Reply by fenske** [reply]
Hi there. It's enough to just create a PR on your repo. The hiring team will then be notified that you've completed your assignment.

**Reply by EmranAhmed00** [reply]
Hi , Thanks for quick reply , I did pull request in my repo . but the task status  did now shows "Complete " its still in Progress , so just confused I  might missed any steps

**Reply by EmranAhmed00** [reply]
did not shows "Complete" i mean

**Reply by fenske** [reply]
Interesting. Let me check.

**Reply by EmranAhmed00** [reply]
Thank you !

**Reply by fenske** [reply]
![CleanShot 2023-08-04 at 21 46 11@2x](https://github.com/DevSkillsHQ/discusssions/assets/1162212/98021eed-4978-4a10-99f3-9bf8a9d44a82)

This part is missing.

**Reply by fenske** [reply]
It seems like this was counter-intuitive somehow. Do you think a "Submit" button in the app would be more clear?

**Reply by EmranAhmed00** [reply]
yeah definitely submit button would more clear and understandable

**Reply by EmranAhmed00** [reply]
i have used this for my task -- fullstack-boilerplate-dotnet-angular--  follow doc same its repo name and "Implementation"  for branch name .  so do i need to create another pull request or !

**Reply by fenske** [reply]
The PR should be created on the repo linked in your Alva application page (https://github.com/ffc1e12/transaction-management-backend-level-1_13989e3-9bc87b).

**Reply by EmranAhmed00** [reply]
Okay. Thank you for the instructions

**Reply by fenske** [reply]
Ofc. Best of luck with the rest of the interview process!

**Reply by EmranAhmed00** [reply]
Thank you so much ! appreciate your quick help .

**Reply by reenhilm** [reply]
I have created such a pull request. I am done now? There is no confirmation. It still says my test is in progress.

**Reply by fenske** [reply]
@reenhilm Which repo have you created it on?

**Reply by reenhilm** [reply]
[kodtestBackend2](https://github.com/reenhilm/kodtestBackend2)

**Reply by fenske** [reply]
Ok, but that is not the repository you received in your Alva invite. 

As it suggests in the README of you repo, you should first complete the task in the description, and then create a PR on the repository linked from your Alva page. 

Hope this clarifies it.

**Reply by reenhilm** [reply]
it clearly says in the instructions:
Go to the "Actions" tab of **your** GitHub repository and select the "Setup boilerplate" workflow in the left side panel.

YOUR as in I should create my own from a choosen boilerplate. It also says your repository later on in the description a second time.
Also says "do not merge it" as in "do not touch our code". I made 98% in the logic test and still this is very unclear to me how you want things. Write more clear instructions? It is still unclear to me how to provide you with my code I have put alot of work on .. I'm used two-three days of my time on this very unclear test

**Reply by fenske** [reply]
Alright, _your repository_ in this context is the one you received from Alva because it is linked to your candidate page. From what I can tell it doesn't have any code submitted nor has a PR open which is why it still say "In progress" on your candidate page.

Here is what you gotta do:

1. Solve the task as described in the README.
2. Push your code to the new `implementation` branch in your repository (as in the one linked from your candidate page in Alva).
3. Create a new pull request on that repo (from `implementation` to `master`, but do not merge it.

That should be it.

**Reply by fenske** [reply]
Alternatively, feel free to attach a zip file with your solution here. I'll submit it on your behalf. 🫡

**Reply by reenhilm** [reply]
zip file will be to big it's 31MB. Can't you just use the "download zip" from the repo [kodtestBackend2](https://github.com/reenhilm/kodtestBackend2)

**Reply by fenske** [reply]
Ok, assuming you didn't want to add anything more there, sure. Created it [here](https://github.com/ffc1e12/transaction-management-backend-level-1_8bd0c8a-hajfb4/pull/1).

I wish you the best of luck with the rest of the interview process!

BR,
Anton.

**Comment by alitanharo** [comment]
Hi there,
 I've already implemented this app and successfully passed the test in another process! Currently, I am completing the same test for another company. I'm utilizing the same codebase, and everything is functioning as required on my local machine.
However, I've encountered an issue when running the "Transaction Management Backend - Level 2" test. This failure is due to the following reasons:
1. The need to create a transaction, read it, and retrieve the updated account balance.
2. The requirement to create transactions with negative amounts.
It's worth noting that my code is fully aligned with the specifications outlined in test.spec.js, and it operates flawlessly on my local machine. Could you provide me with some guidance to address these issues?

**Reply by fenske** [reply]
Hi there. As you can see in the logs, the backend returns a 404 instead of a 201. You can reproduce the error if you run your project via Gitpod (click on this link to spin up one: https://gitpod.io/#https://github.com/ffc1e12/transaction-management-fullstack-level-2_e4430d0-h56ibf).

Here's also a video of a test run to help you troubleshoot.

https://github.com/DevSkillsHQ/discusssions/assets/1162212/b0991598-331c-4b10-930e-541e4a931b21

Let me know if you if I can assist further.

BR,
Anton from Alva.

**Reply by alitanharo** [reply]
Hi Fenske,

I appreciate your prompt response. However, I must admit that I find myself a bit perplexed. As I had previously mentioned, I had already completed the test in question. During my previous attempt, I utilized an external API endpoint to interact with the API. The specific URL endpoint I employed was: https://infra.devskills.app/api/accounting. I can confirm that this approach yielded successful results.

In the present scenario, I am encountering some difficulties. It has come to my attention that you have introduced a different API: https://infra.devskills.app/accounting/api/3.1.0. The format of this API appears to be distinct from what I had encountered earlier. I am uncertain whether I am expected to implement this new API or simply make API calls to it. Moreover, I am unclear about the precise URL endpoint to be used.

Could you kindly provide clarification on this matter? I am confident that my application functions seamlessly, yet I am encountering challenges in passing the Cypress test. Regrettably, I am unsure of the underlying cause of this issue. Your insights and guidance are greatly appreciated.

Thank you once again for your time and consideration.

**Reply by alitanharo** [reply]
![test1](https://github.com/DevSkillsHQ/discusssions/assets/90282929/aa61fcb5-e55d-4fa7-a1d6-a119786b4d3e)

**Reply by fenske** [reply]
Thx for the clarification. I didn't realize the original implementation proxied calls to the API. 

So we're on the same page. The task was actually about you implementing the provided API spec from scratch, not via proxying the calls. It was apparently confusing in the past. That's why we removed the link to our hosted API and instead shared the spec you pointed out.

But given you've now spent quite some time working on this task, it might be ok with the hiring team if you leave it as is. But I'd recommend double-checking with them.

As for the failing Cypress tests, I spun up a Gitpod on your repo by going to https://gitpod.io/#https://github.com/ffc1e12/transaction-management-fullstack-level-2_e4430d0-h56ibf, switched to the implementation branch, and got the same output as in the pipeline. (Manually running queries on the API also produced the wrong output). 

Can you please spin up the Gitpod by clicking on this link so you can also reproduce it?

I can see your screenshot with passed tests, but the workspace path suggests you've spun it up from a different repo called transaction-management-fullstack-level-3_7b05c62-17a6ge. Where did you take it from? 🤔

**Comment by srilakshmivenkatraman** [comment]
Hi, I would like to code with springboot maven application for backend and react for front end.
Where can I find the boiler plate for this?

**Reply by fenske** [reply]
Hi there. We don't have a boilerplate for maven at the moment, but [fullstack-boilerplate-java-springboot-react-typescript](https://github.com/DevSkillsHQ/fullstack-boilerplate-java-springboot-react-typescript) could be easily adapted.

Here's how to do it:
1. Import this boilerplate using the instructions in your project README file.
2. Go to https://start.spring.io/ and generate a spring boot Maven project.
3. Substitute the content of `app-backend` with your newly generated Spring Boot project.
4. Update the `build:backend` and `start:backend` commands that will respectively build and start your Spring Boot project.

Let me know if you have any more questions.

BR,
Anton.

**Comment by ohthepain** [comment]
How should I submit my merge requests? I was not able to use the boilerplates for react and flask so I forked the boilerplate repos. Can I submit MRs to the boilerplate repos? I see that the boilerplate repos don't have the same owner as the test repo.

**Comment by TonyKarlsson** [comment]
My tests passes locally, but when I push, the job "Run tests with waiting on API" fails, saying:
Unable to locate executable file: "null"/ping. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also check the file mode to verify the file is executable.

**Reply by fenske** [reply]
Hi Tony. I pushed a fix to your `implementation` branch that initializes the baseUrl correctly. 

We have fixed the config in the template version of this coding challenge but since you started working on your task before the fix was applied, it wasn't propagated to your repo.

Sorry about this.

I wish you the best of luck with the rest of your interview process!

BR,
Anton from Alva.

**Reply by TonyKarlsson** [reply]
Ah, I see! Thanks for a quick response and fix. I think all is working properly now

**Reply by FThelin** [reply]
I have the same issue. But mine says Unable to locate executable file: "http://localhost:5500"/ping. :/

**Comment by greazleay** [comment]
Hi,

I need a bit of clarification, the instructions says not to implement the POST /presentations and GET /presentations/{presentation_id} routes but proxy the responses from the remote server instead and it also says store your polls and votes locally on an SQLite database. But creating presentations in the remote server requires having polls in the request body which means the polls are also stored in the remote server either way and fetching a presentation by id also returns the presentation with the polls. 

I just wanted to be sure this is the case and I correctly understand the instructions

**Reply by fenske** [reply]
Hi @greazleay. You've got it right! Essentially, we're aiming to establish the logic for navigating through the presentation, conducting votes, and saving the presentation data in an external service. We'd then retrieve the data from this service as needed.

**Reply by greazleay** [reply]
Thanks for the response @fenske

**Comment by yansen0508** [comment]
![environment setting](https://github.com/DevSkillsHQ/discusssions/assets/36081071/2792a67f-bfa5-4431-8371-4ae23f844098)
Hi there,

I participated in the  "irt-param-estimation" test.
I followed the instructions to work in the Gitpod environment (install JetBrains Gateway) and I had the following issue related to the environment configuration.

I need your help :)
SY

**Reply by fenske** [reply]
Hi @yansen0508. Could you please try working in a GitHub Codespace instead?

here's how:
![CleanShot 2023-10-20 at 23 40 54@2x](https://github.com/DevSkillsHQ/discusssions/assets/1162212/7fcbaeb1-7031-4718-9a42-d6c982a7418b)
